pkg/services/grpcserver/query.go

```diff
@@ -0,0 +1,64 @@
+package grpcserver
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/log"
+	grpccontext "github.com/grafana/grafana/pkg/services/grpcserver/context"
+	"github.com/grafana/grafana/pkg/services/query"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type QueryDataService struct {
+	cfg    *setting.Cfg
+	server *queryDataServer
+}
+
+func ProvideQueryDataService(cfg *setting.Cfg, grpcServerProvider Provider, queryService *query.Service) (*QueryDataService, error) {
+	server := &queryDataServer{
+		logger:       log.New("grpc-server-query-data"),
+		queryService: queryService,
+	}
+	pluginv2.RegisterDataServer(grpcServerProvider.GetServer(), server)
```

Comment:
Nit. It's a bit weird to reuse this contract from the plugin developer perspective since you might get the impression you need to populate the full `PluginContext` of the request. In a perfect world the HTTP `/api/ds/query` and gRPC `QueryData` would use similar contracts since it's basically just two different transports. Something to think about moving forward I suppose?